File: src/main/java/io/github/mg138/tsbook/command/Effect.java
package io.github.mg138.tsbook.command;

import io.github.mg138.tsbook.Book;
import io.github.mg138.tsbook.entities.effect.EffectHandler;
import io.github.mg138.tsbook.entities.effect.data.StatusEffectType;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class Effect {
    public static boolean call(CommandSender sender) {
        sender.sendMessage(Book.getCfg().translate.translate("commands.feedback.effect"));
        return true;
    }

    public static boolean call(CommandSender sender, String playerName, String effectName, String literalPower, String literalTicks) {
        Player player = Book.getInst().getServer().getPlayer(playerName);
        if (player == null) {
            sender.sendMessage(Book.getCfg().translate.translate("errors.player_not_found"));
            return false;
        }
        try {
            StatusEffectType type = StatusEffectType.valueOf(effectName.toUpperCase());
            double power = Double.parseDouble(literalPower);
            int ticks = Integer.parseInt(literalTicks);
            EffectHandler.apply(type, player, power, ticks);
            sender.sendMessage(Book.getCfg().translate.translate("messages.effect.applied"));
            return true;
        } catch (NumberFormatException e) {
            sender.sendMessage(Book.getCfg().translate.translate("errors.should_put_number"));
            return false;
        }
    }

    public static boolean clear(CommandSender sender, String playerName) {
        Player player = Book.getInst().getServer().getPlayer(playerName);
        if (player == null) {
            sender.sendMessage(Book.getCfg().translate.translate("errors.player_not_found"));
            return false;
        }
        try {
            EffectHandler.remove(player);
            sender.sendMessage(Book.getCfg().translate.translate("messages.effect.cleared"));
            return true;
        } catch (NullPointerException e) {
            sender.sendMessage(Book.getCfg().translate.translate("errors.effect.no_active_effect"));
            return false;
        }
    }
}

File: src/main/java/io/github/mg138/tsbook/command/admin/Effect.kt
package io.github.mg138.tsbook.command.admin

import io.github.mg138.tsbook.Book.Companion.setting
import io.github.mg138.tsbook.command.util.error.CommandError
import io.github.mg138.tsbook.command.util.CommandUtil
import io.github.mg138.tsbook.command.util.error.CommandEffectError
import io.github.mg138.tsbook.entities.effect.data.StatusEffectType
import io.github.mg138.tsbook.entities.effect.EffectHandler
import java.lang.NumberFormatException
import java.lang.NullPointerException
import org.bukkit.command.CommandSender

object Effect {
    fun call(sender: CommandSender): Boolean {
        sender.sendMessage(setting.translate.translate("commands.feedback.effect"))
        return true
    }

    fun call(
        sender: CommandSender,
        playerName: String,
        effectName: String,
        literalPower: String,
        literalTicks: String
    ): Boolean {
        val player = CommandUtil.getPlayerByName(playerName, sender) ?: return false

        return try {
            val type = StatusEffectType.valueOf(effectName.toUpperCase())
            val power = literalPower.toDouble()
            val ticks = literalTicks.toInt()
            EffectHandler.apply(type, player, power, ticks)
            sender.sendMessage(setting.translate.translate("messages.effect.applied"))
            true
        } catch (e: NumberFormatException) {
            CommandError.shouldPutNumber(sender)
            false
        }
    }

    fun clear(sender: CommandSender, playerName: String): Boolean {
        val player = CommandUtil.getPlayerByName(playerName, sender) ?: return false

        return try {
            EffectHandler.remove(player)
            sender.sendMessage(setting.translate.translate("messages.effect.cleared"))
            true
        } catch (e: NullPointerException) {
            CommandEffectError.noActiveEffect(sender)
            false
        }
    }
}